sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,557 @@
+ * @param[in] remote    Remote session to close
+ */
+void sock_dtls_close_session(sock_dtls_t *sock, sock_dtls_session_t *remote);
+
+/**
+ * @brief Decrypts and reads a message from a remote peer.
+ *
+ * @param[in] sock      DTLS sock to use.
+ * @param[out] remote   Remote DTLS session of the received data.
+ *                      Cannot be NULL.
+ * @param[out] data     Buffer where the data should be stored.
+ * @param[in] maxlen    Maximum memory available at @p data.
+ * @param[in] timeout   Timeout for receive in microseconds.
+ *                      If 0 and no data is available, the function returns
+ *                      immediately.
+ *                      May be SOCK_NO_TIMEOUT to wait until data
+ *                      is available.
+ *
+ * @note Function may block if data not available and @p timeout != 0
+ *
+ * @return The number of bytes received on success
+ * @return value < 0 on error
+ * @return  -EADDRNOTAVAIL, if the local endpoint of @p sock is not set.
+ * @return  -EAGAIN, if @p timeout is `0` and no data is available.
+ * @return  -EINVAL, if @p remote is invalid or @p sock is not properly
+ *          initialized (or closed while sock_udp_recv() blocks).
+ * @return  -ENOBUFS, if buffer space is not large enough to store received
+ *          data.
+ * @return  -ENOMEM, if no memory was available to receive @p data.
+ * @return  -ETIMEDOUT, if @p timeout expired.
+ */
+ssize_t sock_dtls_recv(sock_dtls_t *sock, sock_dtls_session_t *remote,
+                       void *data, size_t maxlen, uint32_t timeout);
+
+/**
+ * @brief Encrypts and sends a message to a remote peer
+ *
+ * @param[in] sock      DTLS sock to use
+ * @param[in] remote    DTLS session to use. A new session will be established
+ *                      if no session exist between client and server.
+ * @param[in] data      Pointer where the data to be send are stored
+ * @param[in] len       Length of @p data to be send
+ *
+ * @note Function may block
+ *
+ * @return The number of bytes sent on success
+ * @return value < 0 on error
+ * @return  -EADDRINUSE, if sock_dtls_t::udp_sock has no local end-point.
+ * @return  -EAFNOSUPPORT, if `remote->ep != NULL` and
+ *          sock_dtls_session_t::ep::family of @p remote is != AF_UNSPEC and
+ *          not supported.
+ * @return  -EHOSTUNREACH, if sock_dtls_session_t::ep of @p remote is not
+ *          reachable.
+ * @return  -EINVAL, if sock_udp_ep_t::addr of @p remote->ep is an
+ *          invalid address.
+ * @return  -EINVAL, if sock_udp_ep_t::port of @p remote->ep is 0.
+ * @return  -ENOMEM, if no memory was available to send @p data.
+ */
+ssize_t sock_dtls_send(sock_dtls_t *sock, sock_dtls_session_t *remote,
+                       const void *data, size_t len);
+
+/**
+ * @brief Destroys DTLS sock created by `sock_dtls_create`
+ *
+ * @pre `(sock != NULL)`
+ *
+ * @param sock          DTLS sock to destroy
+ */
+void sock_dtls_destroy(sock_dtls_t *sock);
```

Comment:
```suggestion
void sock_dtls_close(sock_dtls_t *sock);
```

to be in line with the other `sock` APIs?

Reply:
Makes sense. I'll rename it.